src/library_webgl.js

```diff
@@ -2587,7 +2627,11 @@ var LibraryGL = {
       if (source.indexOf('#version 100') != -1) {
         source = source.replace(/#extension GL_OES_standard_derivatives : enable/g, "");
         source = source.replace(/#extension GL_EXT_shader_texture_lod : enable/g, '');
-        var prelude = '';
+        var prelude =
+#if GL_MAX_FEATURE_LEVEL >= 30
+          GL.currentContext.featureLevel == 30 ? '#version 310 es\n' :
+#endif
+          '#version 300 es\n';
```

Comment:
If the original shader was `#version 100` I don't see any reason to conditionally upgrade it to `#version 310`. `#version 300` should always be okay (you can undo these changes and keep the original code).

Reply:
Ohhh, I thought `#version 300 es` is only for compatibility purpose in `WebGL2ComputeRenderingContext` and therefore it should be discouraged. I'll reconsider this.